akka-stream-tests/src/test/java/akka/stream/javadsl/FlowTest.java

```diff
@@ -739,6 +742,39 @@ public Integer apply(Throwable elem, boolean isCheck) {
     future.toCompletableFuture().get(200, TimeUnit.MILLISECONDS);
   }
 
+  @Test
+  public void mustBeAbleToRecoverTermination() throws Exception {
+    final TestPublisher.ManualProbe<Integer> publisherProbe = TestPublisher.manualProbe(true,system);
+    final TestKit probe = new TestKit(system);
+
+    final Source<Integer, NotUsed> source = Source.fromPublisher(publisherProbe);
+    final Flow<Integer, Integer, NotUsed> flow = Flow.of(Integer.class).map(
+      new Function<Integer, Integer>() {
+        public Integer apply(Integer elem) {
+          if (elem == 2) throw new RuntimeException("ex");
+          else return elem;
+        }
+      })
+      .recoverTermination(new JavaPartialFunction<Try<Done>, Try<Done>>() {
```

Comment:
maybe a total function would be nicer?

Reply:
also in scala I guess?

Reply:
Ehrm, maybe yeah. The result would then look like this:

```scala
.recoverTermination {
  case Failure(_: ExceptionIWantToHandle) => Success(Done)
  case pass => pass
}
```

It's certainly nicer on the java side and for simplicity, let's just do it on the Scala side as well. Makes the behavior more apparent to the user (as-in: what is the default behavior?!) as well.

Reply:
yeah 👍